### ISSUE TITLE
Tensorflow export fails

### ISSUE THREAD

**Issue by NSDCars5** [Cloud-CV/Fabrik#195]
Exporting a model via Tensorflow fails, and the server delivers a 500 Internal Server Error. Not knowing how to deal with this, web IDE delivers no output and stays stuck at loading screen. 

**Expected behavior:** a download of a generated model.

**Observed behavior:** stuck at loading screen, console shows server sent error 500.

**Steps to reproduce:**

1. Add input layer

2. Add softmax layer, join with input layer

3. Export as Caffe, works fine

4. Export as TensorFlow, fails as above

**Server logs:**

```
Traceback (most recent call last):
  File "/home/nsood/Fabrik/tensorflow_app/caffe-tensorflow/convert.py", line 60, in <module>
    main()
  File "/home/nsood/Fabrik/tensorflow_app/caffe-tensorflow/convert.py", line 56, in main
    args.phase)
  File "/home/nsood/Fabrik/tensorflow_app/caffe-tensorflow/convert.py", line 27, in convert
    transformer = TensorFlowTransformer(def_path, caffemodel_path, phase=phase)
  File "/home/nsood/Fabrik/tensorflow_app/caffe-tensorflow/kaffe/tensorflow/transformer.py", line 221, in __init__
    self.load(def_path, data_path, phase)
  File "/home/nsood/Fabrik/tensorflow_app/caffe-tensorflow/kaffe/tensorflow/transformer.py", line 227, in load
    graph = GraphBuilder(def_path, phase).build()
  File "/home/nsood/Fabrik/tensorflow_app/caffe-tensorflow/kaffe/graph.py", line 255, in build
    graph.compute_output_shapes()
  File "/home/nsood/Fabrik/tensorflow_app/caffe-tensorflow/kaffe/graph.py", line 99, in compute_output_shapes
    node.output_shape = TensorShape(*NodeKind.compute_output_shape(node))
TypeError: __new__() takes exactly 5 arguments (2 given)
Internal Server Error: /tensorflow/export
Traceback (most recent call last):
  File "/home/nsood/Fabrik/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/nsood/Fabrik/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/nsood/Fabrik/local/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/nsood/Fabrik/tensorflow_app/views/export_graphdef.py", line 37, in export_to_tensorflow
    input_caffe = map(int, input_dim.split(','))
AttributeError: 'NoneType' object has no attribute 'split'
[18/Dec/2017 17:53:17] "POST /tensorflow/export HTTP/1.1" 500 17961
```

**Comment by utsavgarg** [comment]
Tensorflow support has not been worked on in some time, but we should at least try to make it work properly for the layers we support.

**Comment by PalashTanejaPro** [comment]
After some testing with what @NSDCars5 mentioned here. Here are my observations:
  1. Caffe export works because unlike Tensorflow it doesn't check the validity of the model, it is purely syntactical. 
  2. I am running on my fixed version of the Tensorflow backend in #214 and adding an fc layer before the softmax works for me.

[20180101043919tutmc.pbtxt.txt](https://github.com/Cloud-CV/Fabrik/files/1595609/20180101043919tutmc.pbtxt.txt)

**Comment by utsavgarg** [comment]
Fixed in #214
